libsolidity/ast/Types.cpp

```diff
@@ -961,13 +961,15 @@ BoolResult RationalNumberType::isExplicitlyConvertibleTo(Type const& _convertTo)
 {
 	if (isImplicitlyConvertibleTo(_convertTo))
 		return true;
-	else if (_convertTo.category() != Category::FixedBytes)
+
+	auto category = _convertTo.category();
+	if (category == Category::FixedBytes || (isNegative() && category == Category::Address))
```

Comment:
If this is changed into something like
```suggestion
	if (category == Category::FixedBytes || (isNegative() && isNotSigned(_convertTo))
```
this PR also fixes https://github.com/ethereum/solidity/issues/9861
(assuming that it's supposed to be fixed as well?)

Reply:
Yeah, but that one should be more controversial, since `uint(-1)` etc is common in contracts.

Reply:
The expression `uint(-1)` might be less common since we have `type(uint).max`

Reply:
Yea, or it at least *should* be